### PR TITLE
Baremetal documentation

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
User guide for deploying and provisioning in the bare-metal lab with metal3.

### CODE REVIEW

**Comment by lentzi90** [comment]
Can you add this page in the [summary](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/SUMMARY.md) somewhere so it is possible to find it? For some reason I cannot find it in the preview even when typing in the exact address :sweat_smile:

File: docs/user-guide/src/baremetal/guide.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+# Baremetal provisioning
+
+This is a guide to provision baremetal servers using Metal³ project. This is generic guide with basic implementation, different hardwares may require different configuration.
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:3]:
```suggestion
This is a guide to provision baremetal servers using the Metal³ project. This is generic guide with basic implementation, different hardware may require different configuration.
In this guide we will use minikube as management cluster.
All commands are executed on the host where minikube is set up.
This is a separate machine, e.g. your laptop or one of the servers, that has access to the network where the servers are in order to provision them.
```

File: docs/user-guide/src/baremetal/guide.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  ```
+
+* Launch ironic.
+
+  ```bash
+  # Run deploy.sh
+  ./baremetal-operator/tools/deploy.sh -i -k -t
+  ```
+
+## Create secrets and crds for BareMetalHosts
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:145]: The CRDs were already installed with BMO. I think you meant the CRs, but they are BMHs so lets just call them that :slightly_smiling_face: 
```suggestion
## Create Secrets and BareMetalHosts
```

File: docs/user-guide/src/baremetal/guide.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  Login to the host from where you want to provision. The baremetal nodes should be accesible from the host via one of the following protocols
+
+    * IPMI
+    * Redfish
+    * WSMAN
+    * iRMC
+    * ibmc
+    * iLO
+
+  Here is description of bmh crd <https://github.com/metal3-io/baremetal-operator/blob/main/config/crd/bases/metal3.io_baremetalhosts.yaml>. Create yaml files for each bmh that will be used. Below is an example
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:156]: Not sure if it is useful to link to the definition. It is is certainly not very human readable :slightly_frowning_face: 
```suggestion
Create yaml files for each BareMetalHost that will be used. Below is an example.
```

File: docs/user-guide/src/baremetal/guide.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      online: true
+      bootMACAddress: <<mac_address_bmh1>>
+      bootMode: legacy
+      bmc:
+        address: <<address_bmh1>> // this depends on the protocol that are mentioned above, they depend on hardware vendor
+        credentialsName: <<secret_name_bmh1>>
+        disableCertificateVerification: true
+  ```
+
+  Deploy bmhost yamls.
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:183]:
```suggestion
Apply the manifests.
```

File: docs/user-guide/src/baremetal/guide.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+  ```bash
+  kubectl apply -f ./bmh1.yaml -n metal3
+  ```
+
+  Wait for bmh to be available. Once bmhs are available, we management cluster and provisioned bmhs. Next step is to create target cluster that will use these bmhs.
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:189]:
```suggestion
At this point, the BareMetalHosts will go through `registering` and `inspection` phases before they become `available`.
Wait for all of them to be available. You can check their status with `kubectl get bmh -n metal3`.
The next step is to create a workload cluster from these BareMetalHosts.
```

**Comment by adilGhaffarDev** [comment]
> Can you add this page in the [summary](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/SUMMARY.md) somewhere so it is possible to find it? For some reason I cannot find it in the preview even when typing in the exact address sweat_smile

added

File: docs/user-guide/src/baremetal/guide.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+  # Disables basic authentication for Ironic API. Any value provided in this variable disables authentication.
+  # To enable authentication, leave this variable unset. If unset, then IRONIC_USERNAME and IRONIC_PASSWORD must be set.
+  export IRONIC_NO_BASIC_AUTH=true
+
+  # Disables basic authentication for Ironic inspector API. Any value provided in this variable disables authentication.
+  # To enable authentication, leave this variable unset. If unset, then IRONIC_INSPECTOR_USERNAME and IRONIC_INSPECTOR_PASSWORD must be set.
+  export IRONIC_INSPECTOR_NO_BASIC_AUTH=true
+
+  # Generate templates with clusterctl, change control plane and worker count according to the number of bmhs
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:218]: It seems the image related variables are missing above?

**Review by lentzi90 — COMMENTED** [review]
Thanks for this! It will be very useful I believe!

I have added quite a lot of suggestions below. We don't have to go through all of it before taking this in though. Feel free to skip some and we can handle them later.

In general I just wanted to point out that it would be good to avoid too much short acronyms and just spell out the full word in guides like there.

File: docs/user-guide/src/SUMMARY.md [hunk 7]
```diff
@@ -29,6 +29,7 @@
 - [Ip-address-manager](ipam/introduction.md)
     - [Install Ip-address-manager](ipam/ipam_installation.md)
 - [Developer environment]()
+- [Baremtal](baremetal/guide.md)
```
Comment (jayofdoom) [inline docs/user-guide/src/SUMMARY.md:32]:
```suggestion
- [Baremetal](baremetal/guide.md)
```

**Comment by adilGhaffarDev** [comment]
@lentzi90 I have added the suggested changes. Kindly check.

File: docs/user-guide/src/baremetal/guide.md [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+* Start httpd container. This is used to host the the OS images that the BareMetalHosts will be provisioned with.
+
+  ```bash
+  sudo docker run -d --net host --privileged --name httpd-infra -v /opt/metal3-dev-env/ironic:/shared --entrypoint /bin/runhttpd --env
+  ```
+
+  Download image from here <https://artifactory.nordix.org/artifactory/metal3/images/k8s_v1.27.1> and put it /opt/metal3-dev-env/ironic/html/images folder.
+
+  Convert qcow2 image to raw image and create hash of raw image
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:111]:
```suggestion
  Convert the qcow2 image to raw format and get the hash of the raw image
```

File: docs/user-guide/src/baremetal/guide.md [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+
+  # Disables basic authentication for Ironic inspector API. Any value provided in this variable disables authentication.
+  # To enable authentication, leave this variable unset. If unset, then IRONIC_INSPECTOR_USERNAME and IRONIC_INSPECTOR_PASSWORD must be set.
+  export IRONIC_INSPECTOR_NO_BASIC_AUTH=true
+
+  # Export node image variable and node image hash varibale that we created before. Change name according to what was downlowded from artifactory
+  export IMAGE_RAW_URL=http://172.22.0.1/images/CENTOS_9_NODE_IMAGE_K8S_v1.27.1-raw.img
+  export IMAGE_RAW_CHECKSUM=http://172.22.0.1/images/CENTOS_9_NODE_IMAGE_K8S_v1.27.1-raw.img.sha256sum
+
+  # Generate templates with clusterctl, change control plane and worker count according to the number of bmhs
```
Comment (lentzi90) [inline docs/user-guide/src/baremetal/guide.md:233]: Could you wrap the comments so we don't get so long lines? In the web page preview these go outside the code box so it is necessary to scroll to see them.

**Review by lentzi90 — COMMENTED** [review]
Thanks for this! No more comments from me.

/approve

**Comment by lentzi90** [comment]
No approvers rights in the user guide I see :disappointed: 
Well I will put lgtm then :triumph:
/lgtm

**Comment by elfosardo** [comment]
/approve
thanks

**Pull request merged**